client/src/pages/signup.js

```diff
@@ -0,0 +1,279 @@
+        width: "20rem",
+        maxWidth: "40vw"
+    },
+    formControl: {
+        paddingTop: "1rem",
+        paddingBottom: "1rem"
+    },
+    toSignIn: {
+        display: "flex",
+        alignSelf: "flex-end",
+        alignItems: "center",
+        marginTop: "10px",
+        marginRight: "1rem",
+        "& Button": {
+            marginLeft: "10px",
+            padding: ".2rem",
+            borderRadius: "50px",
+            width: "100px",
+            height: "25px"
+        }
+    },
+    rightScreen: {
+        display: "flex",
+        flexDirection: "column",
+        width: "700px"
+    },
+    submitButton: {
+        marginTop: "1rem",
+        padding: ".5rem",
+        borderRadius: "50px",
+        width: "35%"
+    }
+})
+
+export default function Signup() {
+    const [inputFields, setInputFields] = useState([])
+    const [errors, setErrors] = useState([])
+    const [redirect, setRedirect] = useState(false)
+
+    const classes = useStyles();
+
+    const handleSubmit = (e) => {
+        e.preventDefault();
+
+        // Below sets Errors to a boolean in order to set the error value in the TextField components
+        if (!inputFields.firstName) {
+            setErrors(errors => ({
+                ...errors,
+                firstName: true
+            }));
+        } else {
+            setErrors(errors => ({
+                ...errors,
+                firstName: false
+            }));
+        }
+
+        if (!inputFields.lastName) {
+            setErrors(errors => ({
+                ...errors,
+                lastName: true
+            }));
+        } else {
+            setErrors(errors => ({
+                ...errors,
+                lastName: false
+            }));
+        }
+
+        if (inputFields.email.match(/^([\w.%+-]+)@([\w-]+\.)+([\w]{2,})$/i)) {
+            setErrors(errors => ({
+                ...errors,
+                email: false
+            }));
+        } else {
+            setErrors(errors => ({
+                ...errors,
+                email: true
+            }));
+        }
+
+        if (inputFields.password !== inputFields.confirmPassword) {
+            setErrors(errors => ({
+                ...errors,
+                password: true,
+                confirmPassword: true
+            }));
+        } else {
+            setErrors(errors => ({
+                ...errors,
+                password: false,
+                confirmPassword: false
+                // The request will only be verified/sent after Confirm Password has been entered correctly.
+            }), sendRequest(e));
+        }
+    }
+
+    const sendRequest = (e) => {
+        e.preventDefault()
+        for (var i in errors) {
+            if (errors[i] === true) {
+                console.log("Not all fields filled in")
+                var error;
+                return false;
+            }
+        }
+        // Only sends if all errors are false
+        if (!error) {
```

Comment:
just setting `const error` will still be undefined. So you should be setting `const error = true;`

Reply:
When I set this to const error = true, the if-then statement that follows it cannot read its value. It seems to only read it if it is set to var. What can I do?

Reply:
Actually, I think I solved it. I changed it to a let, and moved the error back one step in the scope, so that the let and the if-then can access it.